bugsnag-android-core/src/main/java/com/bugsnag/android/ErrorReader.java

```diff
@@ -496,13 +533,14 @@ private static CachedThread readThread(JsonReader reader) throws IOException {
     }
 
     @SuppressWarnings("unchecked")
-    private static <T> T deserializeNumber(JsonReader reader) throws IOException {
+    @VisibleForTesting
+    static <T> T deserializeNumber(JsonReader reader) throws IOException {
         try {
             return (T)(Integer) reader.nextInt();
-        } catch (NumberFormatException ex) {
+        } catch (NumberFormatException | IllegalStateException ex) {
```

Comment:
The `JsonReader` GSON implementation can throw an `IllegalStateException` also

Reply:
> Throws IllegalStateException if the next token is not a literal value.

If its not a literal value, none of these options will be valid. `IllegalStateException` should not be in this `catch` clause.

Reply:
This change is required to pass existing tests, specifically `RxErrorTest` which fails to serialize `metaData.device.screenDensity`: https://github.com/bugsnag/bugsnag-android/blob/23f2b4d25a997b5250dd6b371792cef9e8eeb19e/bugsnag-android-core/src/androidTest/java/com/bugsnag/android/RxErrorTest.kt#L49, and `ErrorReaderNumberTest` which fails to read `lineNumber` for stackframe. This version of GSON appears to throw an `IllegalStateException` if a double is supplied when `nextLong()` is called.